### PR TITLE
Feature/fix wallet bonding denom error

### PR DESCRIPTION
**Pull request by mfahampshire**

### CODE REVIEW

File: clients/native/examples/js-examples/websocket/src/index.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ function handleResponse(resp) {
             ourAddress = response.address;
             display("Our address is:  " + ourAddress + ", we will now send messages to ourself.");
         } else if (response.type == "received") {
-            handleReceivedTextMessage(response)
+            // handleReceivedTextMessage(response)
```
Comment (jstuczyn) [inline clients/native/examples/js-examples/websocket/src/index.js]: What happened to `handleReceivedTextMessage` function? If it's unused, just remove it

File: clients/webassembly/js-example/index.js [hunk 2]
```diff
@@ -24,10 +24,16 @@ async function main() {
     // sets up better stack traces in case of in-rust panics
     set_panic_hook();
 
-    // validator server we will use to get topology from
-    const validator = "http://testnet-validator1.nymtech.net:8081"; //"http://localhost:8081";
+    // validator we will use to get topology from
+    // If you run into CORS errors make sure to set `enable-unsafe-cors` as true
+    // and add `'Access-Control-Allow-Origin'` to the accepted headers
+    // in your validator's `config.toml`,
+    const validator = "http://mixnet.club:1317"; //  1317 = standard API port.
```
Comment (jstuczyn) [inline clients/webassembly/js-example/index.js]: you should probably be using different validator - this one is not running milhon

File: clients/webassembly/js-example/index.js [hunk 3]
```diff
@@ -24,10 +24,16 @@ async function main() {
...
 
-    client = new NymClient(validator);
+    // mixnet contract address
+    const mixnet_contract = "hal1k0jntykt7e4g3y88ltc60czgjuqdy4c9c6gv94"
```
Comment (jstuczyn) [inline clients/webassembly/js-example/index.js]: It's finney contract?

File: wallet-web/components/bond/BondNodeForm.tsx [hunk 4]
```diff
@@ -7,8 +7,8 @@ import semver from "semver"
 import { NodeType } from "../../common/node";
 import { theme } from "../../lib/theme";
 import { basicRawCoinValueValidation, makeBasicStyle, validateRawPort } from "../../common/helpers";
-import { Coin, printableCoin } from "@nymproject/nym-validator-client";
-import { DENOM } from "../../pages/_app";
+import { Coin, printableCoin, nativeToPrintable } from "@nymproject/nym-validator-client";
+import { DENOM, UDENOM } from "../../pages/_app";
```
Comment (jstuczyn) [inline wallet-web/components/bond/BondNodeForm.tsx]: No changes in this file seem to use `UDENOM`, so why was it imported?

File: clients/webassembly/js-example/index.js (hunk 3 above)
Comment (mfahampshire) [reply clients/webassembly/js-example/index.js]: this and some other stuff is from when i was messing around with stuff for the 'game' a while back - i'll change it all regardless when i update :)

File: clients/webassembly/js-example/index.js (hunk 2 above)
Comment (mfahampshire) [reply clients/webassembly/js-example/index.js]: this way of getting the topology is now out of date but changed it anyway - should we handle the new example in a different pull?

Comment (futurechimp) [reply clients/webassembly/js-example/index.js]: I think it's reasonable to fix it in this pull.

File: clients/webassembly/js-example/index.html [hunk 5]
```diff
@@ -15,8 +15,17 @@
...
   </p>
-  <p>
+  <!-- <p>
     <label>Message: </label><input type="text" id="message" value="Hello mixnet!">
+  </p> -->
+  <p>
+    <label>Telegram @: </label><input type="text" id="telegram" value="">
+  </p>
+  <p>
+    <label>HAL address: </label><input type="text" id="hal_address" value="">
```
Comment (futurechimp) [inline clients/webassembly/js-example/index.html:25]: This is one line I wonder about - if this example is for milhon, is it not the case that the currency would be more punk? 

There are a couple HAL-related things in the code that I expect would need to change.

File: clients/webassembly/js-example/index.js [hunk 6]
```diff
@@ -51,31 +57,55 @@ async function main() {
...
+ *
  * Message and recipient are taken from the values in the user interface.
  *
  * @param {Client} nymClient the nym client to use for message sending
  */
 async function sendMessageTo() {
-    const message = document.getElementById("message").value;
+    // TODO add input validation of required length for the address & check other two aren't blank before sending
+    const telegram = document.getElementById("telegram").value; // check isnt empty + check first char is @
+    const address = document.getElementById("hal_address").value; // check .length == 42
```
Comment (futurechimp) [inline clients/webassembly/js-example/index.js:68]: HAL-related

File: clients/webassembly/js-example/index.js [hunk 7]
```diff
@@ -51,31 +57,55 @@ async function main() {
...
 /**
  * Display messages that have been sent up the websocket. Colours them blue.
  *
+ * @param {string} telegram
+ * @param {string} hal_address
  * @param {string} message
+ * @param {string} self_address
  */
-function displaySend(message) {
+function displaySend(telegram, hal_address, message, self_address) {
```
Comment (futurechimp) [inline clients/webassembly/js-example/index.js:92]: Hal-related

File: clients/webassembly/js-example/index.js [hunk 8]
```diff
@@ -51,31 +57,55 @@ async function main() {
...
     let paragraph = document.createElement("p")
     paragraph.setAttribute('style', 'color: blue')
-    let paragraphContent = document.createTextNode(timestamp + " sent >>> " + message)
-    paragraph.appendChild(paragraphContent)
+
+
+    let paragraphContent = document.createTextNode(
+      timestamp + " sent the following information >>> " +
+      ' | telegram: ' + telegram +
+      ' | HAL address: ' + hal_address +
```
Comment (futurechimp) [inline clients/webassembly/js-example/index.js:103]: Hal-related

**Review by futurechimp — COMMENTED** [review]
The title of this pull is a bit confusing - I was expecting to see some wallet-related code that fixed the upunk display bug, but this is all examples js and Rust code?

**Comment by mfahampshire** [comment]
@futurechimp there were accidental changes to the client code left in my develop branch when i pushed - the bug is fixed in `wallet-web/components/bond/BondNodeForm.tsx`. I'm happy to close this and just put in another one with only those changes for ease right now.

**Comment by futurechimp** [comment]
Sounds good!

**Pull request closed without merging**